File: src/my_cpp_pkg/src/number_counter.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"
#include "example_interfaces/srv/set_bool.hpp"

using std::placeholders::_1;
using std::placeholders::_2;


class NumberCounterNode : public rclcpp::Node
{
public:
    NumberCounterNode() : Node("number_counter"), sum_(0)
    {
        subscription_ = this->create_subscription<example_interfaces::msg::Int64>("number", 10,
                                                                                  std::bind(&NumberCounterNode::subCallback, this, std::placeholders::_1));
        pub_ = this->create_publisher<example_interfaces::msg::Int64>("number_count", 10);

        server_ = this->create_service<example_interfaces::srv::SetBool>("reset_counter", std::bind(&NumberCounterNode::callbackResetCounter,this, _1, _2));

        RCLCPP_INFO(this->get_logger(), "number_counter has been started");
    }

private:
    void subCallback(const example_interfaces::msg::Int64::SharedPtr msg)   
    {
        sum_ += msg->data;
        auto pubMsg = example_interfaces::msg::Int64();
        pubMsg.data = sum_;
        pub_->publish(pubMsg);
        RCLCPP_INFO(this->get_logger(), "Sum:%ld", pubMsg.data);
    };

    void callbackResetCounter(const example_interfaces::srv::SetBool::Request::SharedPtr request,
                              const example_interfaces::srv::SetBool::Response::SharedPtr response)
    {
        if (request->data == true)
        {
            sum_ = 0;
            response->success = true;
            response->message = "Counter has been reseted";
        }
        else
        {
            response->success = false;
            response->message = "failed to reset counter";
        };
    };

    std::int64_t sum_;
    rclcpp::Subscription<example_interfaces::msg::Int64>::SharedPtr subscription_;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr pub_;
    rclcpp::Service<example_interfaces::srv::SetBool>::SharedPtr server_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberCounterNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/number_counter.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64

class NumberCounterNode(Node):
    def __init__(self):
        super().__init__("number_counter")
        self.create_subscription(Int64,"number",self.subCallback,10)
        self.pub_ = self.create_publisher(Int64,"number_counter",10)
        self.sum_ =0

        self.get_logger().info("number_counter has been started")

    def subCallback(self,msg:Int64):
        self.sum_ += msg.data
        pubMsg = Int64()
        pubMsg.data = self.sum_
        self.pub_.publish(pubMsg)
        


def main(args=None):
    rclpy.init(args=args)
    node = NumberCounterNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()